HomeFinder/HelpClass/HomeComparer.cs

```diff
@@ -0,0 +1,52 @@
+﻿using HomeFinder.Models;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace HomeFinder.HelpClass
+{
+    public class HomeComparer : IComparer<Home>
+    {
+        
+        //Class used with Sort to sort Homes in reverse order by GeneralGrade
+        public int Compare([AllowNull] Home x, [AllowNull] Home y)
+        {
+            if (x.GeneralGrade == null)
+            {
+                if (y.GeneralGrade == null)
+                {
+                    return 0;
+                }
+                else
+                {
+                    return 1;
+                }
+            }
+            else
+            {
+                if (y.GeneralGrade == null)
+                {
+                    return -1;
+                }
+                else
+                {
+                    double difference = (double)x.GeneralGrade - (double)y.GeneralGrade;
```

Comment:
difference between what? same here, this variable should have a better name

Reply:
Ok. I'll change the name to gradeDifference